astropy/coordinates/geodetic.py

```diff
@@ -0,0 +1,107 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import erfa
+
+from astropy import units as u
+from astropy.utils.decorators import format_doc
+
+from .angles import Latitude, Longitude
+from .representation import (
+    BaseRepresentation,
+    CartesianRepresentation,
+)
+
+__all__ = [
+    "BaseGeodeticRepresentation",
+    "StandardGeodeticRepresentation",
+]
+
+ELLIPSOIDS = {}
+"""Available ellipsoids (defined in erfam.h, with numbers exposed in erfa)."""
+# Note: they get filled by the creation of the geodetic classes.
+
+
+def _check_ellipsoid(ellipsoid=None, default="WGS84"):
+    if ellipsoid is None:
+        ellipsoid = default
+    if ellipsoid not in ELLIPSOIDS:
+        raise ValueError(f"Ellipsoid {ellipsoid} not among known ones ({ELLIPSOIDS})")
+    return ellipsoid
+
+
+geodetic_base_doc = """{__doc__}
+
+    Parameters
+    ----------
+    lon, lat : angle-like
+        The longitude and latitude of the point(s), in angular units. The
+        latitude should be between -90 and 90 degrees, and the longitude will
+        be wrapped to an angle between 0 and 360 degrees. These can also be
+        instances of `~astropy.coordinates.Angle` and either
+        `~astropy.coordinates.Longitude` not `~astropy.coordinates.Latitude`,
+        depending on the parameter.
+
+    height : `~astropy.units.Quantity` ['length']
+        The height to the point(s).
+
+    copy : bool, optional
+        If `True` (default), arrays will be copied. If `False`, arrays will
+        be references, though possibly broadcast to ensure matching shapes.
+"""
+
+
+@format_doc(geodetic_base_doc)
+class BaseGeodeticRepresentation(BaseRepresentation):
```

Comment:
I think we should have a more general name, since *geo*detic is for the Earth. How about `BasePlanetodeticRepresentation` (from [wikipedia](https://en.wikipedia.org/wiki/Planetary_coordinate_system)). 

Of course, this also begs the question of how the module should be called....